consul/catalog_endpoint.go

```diff
@@ -159,16 +172,26 @@ func (c *Catalog) ServiceNodes(args *structs.ServiceSpecificRequest, reply *stru
 	}
 
 	// Get the nodes
-	state := c.srv.fsm.State()
-	err := c.srv.blockingRPC(&args.QueryOptions,
+	state := c.srv.fsm.StateNew()
+	err := c.srv.blockingRPCNew(
+		&args.QueryOptions,
 		&reply.QueryMeta,
-		state.QueryTables("ServiceNodes"),
+		state_store.NewMultiWatch(
```

Comment:
I'm actually going to push this down and keep something similar to the old interface (except for KV). This will be a better abstraction and will keep us from needing the state import in several places. Should be an easy change.